returnn/util/basic.py

```diff
@@ -3563,7 +3563,13 @@ def _maybe_compile_inner(self):
       f.write(self.code)
     common_opts = ["-shared", "-O2"]
     if self.is_cpp:
-      common_opts += ["-std=c++14"]
+      cpp_version_opt = "-std=c++14"
+      if BackendEngine.is_tensorflow_selected():
```

Comment:
This code here should be generic and not TF dependent. It should not matter here whether TF is selected, and what GCC was used for TF. This `NativeCodeCompiler` could be used for anything, not just for compiling TF ops.

Maybe we don't add `-std=...` here at all and only add that in `OpCodeCompiler` and `TFNativeUtilCompiler` (which are both TF specific, so there this logic is fine). For that, we might need to extend `common_opts`, e.g. like:
```
common_opts += self._extra_common_opts()
```
And `_extra_common_opts` can be derived and then return the `-std=...` flag.

Alternatively, you might also check `_get_compiler_bin()` here and try to parse the version from it. It is likely sth like `/usr/bin/gcc-4.8` or so.